services/service-runner/test/service-runner.ts

```diff
@@ -57,12 +77,114 @@ describe('Service Runner', function () {
     });
   });
 
+  describe('_getStacCatalogs()', function () {
+
+    const workItemWithOneCatalog = 's3://stac-catalogs/abc/789/outputs/';
+    const workItemWithMultipleCatalogs = 's3://stac-catalogs/abc/321/outputs/';
+    const workItemWithMultipleCatalogsNoBatchFile = 's3://stac-catalogs/abc/987/outputs/';
+    const emptyCatalogUrl = 's3://stac-catalogs/empty/';
+
+    describe('when the directory has no catalogs', async function () {
+      it('returns any empty list', async function () {
+        const files = await _getStacCatalogs(emptyCatalogUrl);
+        expect(files).to.eql([]);
+      });
+    });
+
+    describe('when there is a batch-catalogs.json file associated with the WorkItem', async function () {
+      before(async function () {
+        const s3 = objectStoreForProtocol('s3');
+        const batchFileContent = JSON.stringify([
+          'catalog0.json',
+          'catalog1.json',
+          'catalog2.json',
+          'catalog3.json',
+          'catalog4.json',
+          'catalog5.json',
+          'catalog6.json',
+          'catalog7.json',
+          'catalog8.json',
+          'catalog9.json',
+          'catalog10.json',
+        ]);
+        const stacCatalogsUrl = resolve(workItemWithMultipleCatalogs, 'batch-catalogs.json');
+        await s3.upload(batchFileContent, stacCatalogsUrl, null, 'application/json');
+      });
+
+      it('returns the stac catalogs from batch-catalogs.json in the order they are in the file', async function () {
+        const stacCatalogs = await _getStacCatalogs(workItemWithMultipleCatalogs);
+        expect(stacCatalogs).to.deep.equal([
+          's3://stac-catalogs/abc/321/outputs/catalog0.json',
+          's3://stac-catalogs/abc/321/outputs/catalog1.json',
+          's3://stac-catalogs/abc/321/outputs/catalog2.json',
+          's3://stac-catalogs/abc/321/outputs/catalog3.json',
+          's3://stac-catalogs/abc/321/outputs/catalog4.json',
+          's3://stac-catalogs/abc/321/outputs/catalog5.json',
+          's3://stac-catalogs/abc/321/outputs/catalog6.json',
+          's3://stac-catalogs/abc/321/outputs/catalog7.json',
+          's3://stac-catalogs/abc/321/outputs/catalog8.json',
+          's3://stac-catalogs/abc/321/outputs/catalog9.json',
+          's3://stac-catalogs/abc/321/outputs/catalog10.json',
+        ]);
+      });
+    });
+
+    describe('when there is no batch-catalogs.json file associated with the WorkItem', async function () {
+      describe('when there is just one catalog', async function () {
+        before(async function () {
+          const s3 = objectStoreForProtocol('s3');
+          const catalogContent = JSON.stringify(dummyCatalog);
+          const stacCatalogUrl = resolve(workItemWithOneCatalog, 'catalog.json');
```

Comment:
You are testing the scenario I was worried about so looks like it works.

Reply:
yep